static/app/components/profiling/flamegraphOptionsContextMenu.tsx

```diff
@@ -70,6 +73,25 @@ export function FlamegraphOptionsContextMenu(props: FlameGraphOptionsContextMenu
             >
               {t('Highlight all occurrences')}
             </ProfilingContextMenuItemCheckbox>
+            <ProfilingContextMenuItemButton
+              {...props.contextMenu.getMenuItemProps({
+                onClick: () => {
+                  props.onCopyFunctionNameClick();
+                  // This is a button, so close the context menu.
+                  props.contextMenu.setOpen(false);
+                },
+              })}
+              onClick={e => {
+                // We need to prevent the click from propagating to the context menu layer.
+                e.preventDefault();
+                props.onCopyFunctionNameClick();
+                // This is a button, so close the context menu.
+                props.contextMenu.setOpen(false);
+              }}
```

Comment:
Why do you need to essentially repeat the onClick here? Can the `getMenuItemProps` not return a `onClick` handler like it does with other handlers?

Reply:
I need to double check, for the getMenuItemProps I know we pass the onClick because click can be fired via keyboard navigation and passing it there ensures it is invoked. It does seem unnecessary though, we could just use both